sycl/include/sycl/accessor.hpp

```diff
@@ -414,6 +436,28 @@ constexpr access::target deduceAccessTarget(access::target defaultTarget) {
     return access::target::constant_buffer;
   }
 
+  if constexpr (
+      std::is_same<MayBeTag1,
+                   mode_target_tag_t<access::mode::read,
+                                     access::target::host_task>>::value ||
+      std::is_same<MayBeTag2,
+                   mode_target_tag_t<access::mode::read,
+                                     access::target::host_task>>::value ||
+      std::is_same<MayBeTag1,
+                   mode_target_tag_t<access::mode::write,
+                                     access::target::host_task>>::value ||
+      std::is_same<MayBeTag2,
+                   mode_target_tag_t<access::mode::write,
+                                     access::target::host_task>>::value ||
+      std::is_same<MayBeTag1,
+                   mode_target_tag_t<access::mode::read_write,
+                                     access::target::host_task>>::value ||
+      std::is_same<MayBeTag2,
+                   mode_target_tag_t<access::mode::read_write,
+                                     access::target::host_task>>::value) {
+    return access::target::host_task;
+  }
```

Comment:
Eventually I would love to rewrite this as trait/function utilizing std::optional so we don't need to write out all combinations. That said, you're following the existing implementation so I will not ask it of you.